gamification/src/main/java/com/appcoins/wallet/gamification/repository/BdsPromotionsRepository.kt

```diff
@@ -86,29 +125,40 @@ class BdsPromotionsRepository(private val api: GamificationApi,
     return ForecastBonus(ForecastBonus.Status.INACTIVE)
   }
 
-  override fun getGamificationStats(wallet: String): Single<GamificationStats> {
+  override fun getGamificationStats(wallet: String): Observable<GamificationStats> {
     return getUserStats(wallet)
-        .map { mapToGamificationStats(it) }
+        .map {
+          mapToGamificationStats(it)
+        }
         .flatMap {
           local.setGamificationLevel(it.level)
               .toSingle { it }
+              .toObservable()
```

Comment:
Not on you, but remove the completable from action from the setGamification method and use it on a doOnSuccess, if it's really needed to be a Completable than do a Completable.fromAction on the method that calls it. This way we avoid doing toSingle and toObservable everytime